proposals/schism-eval-fib/README.md

```diff
@@ -0,0 +1,204 @@
+# Schism eval('fib(25)') benchmark
+
+## Overview
+
+This benchmark runs a Scheme interpreter (`eval`), compiled by the
+[Schism](https://github.com/google/schism) self-hosted Scheme
+implementation.  The interpreter is used to compute and print the 25th
+fibonacci number.
+
+This branch of Schism represents Scheme values using `anyref` values
+originating in the embedder.  The embedder has to supply allocator, type
+predicate, and field access instructions for the different Scheme value
+types supported by Schism: small integers, characters, strings, symbols,
+pairs, and closures.  Schism provides a run-time written in JavaScript
+which uses tagged JavaScript numbers for Scheme integers and characters,
+tagged JavaScript strings for Scheme strings and symbols, and JavaScript
+objects for pairs and closures.
+
+This benchmark largely tests the performance of `anyref`, host-side
+allocation and GC, and JavaScript strings.  It also stress-tests calls
+between WebAssembly and JS, as well as inter-WebAssembly calls.
+
+## Possible benchmark evolution
+
+The benchmark could be made more suitable in a few ways:
+
+ * Eq-able reference types would be useful.  It would be nice to be able
+   to compare anyref values by identity, though what that means for host
+   strings is a question.
+
+ * We could use tail calls.
+
+ * `i31ref` would be really useful.
+
+ * The Schism compiler can improve in many ways.  Notably the current
+   version doesn't support static data, which means that it allocates
+   more than usual at run-time in order to create literals anew each
+   time they appear.  We could take a version of `eval` compiled from a
+   newer version of Schism that doesn't have this problem.
+
+ * We could use GC types to avoid calling out so much to the host.
+   Would need implementation of GC in an engine though.
+
+## Benchmark kind
```

Comment:
What about "category" instead of kind?

Reply:
Sure.  FWIW I imagine at some point that there will be a centralized list of benchmarks and that a PR would add the benchmark to the right category on the centralized list.